src/main/java/org/jabref/model/metadata/SaveOrderConfig.java

```diff
@@ -40,15 +66,21 @@ private SaveOrderConfig(List<String> data) {
             throw new IllegalArgumentException();
         }
 
-        String choice = data.get(0);
-        if (ORIGINAL.equals(choice)) {
-            setSaveInOriginalOrder();
-        } else {
-            setSaveInSpecifiedOrder();
+        try {
+            this.orderType = OrderType.valueOf(data.get(0));
+        } catch (IllegalArgumentException ex) {
+            if (data.size() > 1 && data.size() % 2 == 1) {
+                LOGGER.warn("Could not parse sort order: {} - trying to parse the sort criteria", data.get(0));
+                this.orderType = OrderType.SPECIFIED;
+            } else {
+                LOGGER.warn("Could not parse sort order: {}", data.get(0));
+                this.orderType = OrderType.ORIGINAL;
+                return;
+            }
         }
 
         for (int index = 1; index < data.size(); index = index + 2) {
-            sortCriteria.addLast(new SortCriterion(FieldFactory.parseField(data.get(index)), data.get(index + 1)));
+            sortCriteria.add(new SortCriterion(FieldFactory.parseField(data.get(index)), data.get(index + 1)));
```

Comment:
Why did this change?

Reply:
@koppor suggested, that i should switch from LinkedLists to ArrayLists. ArrayLists do not provide a method addLast, but only add, which should add the elements at the end of the list anyway.